service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -205,8 +206,20 @@ public ResponseEntity<PostMethodResponse> postMethod(PostMethodRequest postMetho
         String organization = githubUrlComponents.org();
         branchOrTagName = githubUrlComponents.branchOrTag();
 
-        String githubToken = ecmService.getAccessToken();
-        Boolean isPrivate = gitHubService.isRepoPrivate(organization, repository, githubToken);
+        String githubToken;
+        Boolean isPrivate;
+
+        try {
+          isPrivate = gitHubService.isRepoPrivate(organization, repository, "");
+        } catch (GitHubClient.GitHubClientException e) {
+          githubToken = ecmService.getAccessToken();
+          isPrivate = gitHubService.isRepoPrivate(organization, repository, githubToken);
+        } catch (RestClientException e) {
+          recordMethodCreationCompletion(
+              methodSource, HttpStatus.NOT_FOUND.value(), requestStartNanos);
+          return new ResponseEntity<>(
+              new PostMethodResponse().error(e.getMessage()), HttpStatus.NOT_FOUND);
+        }
```

Comment:
It seems we are catching general `RestClientException` here but throwing 404 Not found to the user. We should throw the response code that is returned in the exception

Reply:
Ah, thank you for catching. I switched to this exception since I see that's what's thrown when `.getAccesToken()` fails. Since this is a general exception, I'm giving it the 500 status, similar to the block of exceptions we have below.